XivelySDKAndroid/src/main/java/com/xively/internal/rest/timeseries/TimesSeriesApiFactory.java

```diff
@@ -0,0 +1,56 @@
+package com.xively.internal.rest.timeseries;
+
+import com.xively.XiSdkConfig;
+import com.xively.internal.Config;
+import com.xively.internal.logger.LMILog;
+import com.xively.sdk.BuildConfig;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+public class TimesSeriesApiFactory {
+    private GetData timeseriesApi;
+
+    private TimesSeriesApiFactory() {
```

Comment:
Could not we make these Factories a common class?
I saw exactly the same ~50 lines for every service.

Reply:
I've cleaned out a small part of it but right now I cannot give a generic solution because of the dynamic amount of API creation in each constructor especially in BP. :/